packages/machine/src/instruction-executor.ts

```diff
@@ -64,46 +65,64 @@ export class InstructionExecutor implements Observable {
   public buildExecutionsFromLog(log: Log): ActionExecution[] {
     return Object.keys(log).map(key => {
       const entry = log[key];
-      const action = new Action(
-        entry.requestId,
-        entry.actionName,
-        entry.clientMessage,
-        entry.isAckSide
-      );
       const execution = new ActionExecution(
-        action,
+        entry.actionName,
```

Comment:
@IIIIllllIIIIllllIIIIllllIIIIllllIIIIll Would it be better to maybe supply a configuration object? This constructor has grown to have too many arguments. Maybe we can do something like:

```ts
interface ActionExecutionConfiguration {
    actionName: cf.legacy.node.ActionName,
    instruction: Opcode,	    
    instructions: Opcode[],
    instructionPointer: number,
    clientMessage: cf.legacy.node.ClientActionMessage,	    
    instructionExecutor: InstructionExecutor,
    results: OpCodeResult[] = []	    
    isAckSide: boolean,
    requestId: string,
    intermediateResults: IntermediateResults
}
```

And then pass an object to the constructor:

```ts
const execution = new ActionExecution({
  requestId: '',
  // other properties
});
```

Notice that you do not need to instantiate `ActionExecutionConfiguration`. `tsc` itself will match the object structure to the interface. 

This would leave you with the following declaration:

```ts
class ActionExecution {
  constructor(config: ActionExecutionConfiguration) { 
     // Stuff happens here.
  }
}
```

Reply:
we can remove `entry.instructionPointer`, `entry.isAckSide` and `entry.requestId` and `entry.intermediateResults` in subsequent PRs. I don't like config objects as they can be passed around and hide classes with large sets of parameters.

Reply:
Fair enough, but I do believe that a constructor with 10 arguments (!) is a bit too much :P If we can minimize the signature as much as possible, our future selves will be thankful. ^^

Reply:
It's actually only 8...and we can get it down to 4 easily...

```
    actionName: cf.legacy.node.ActionName,
    instructions: Opcode[],
    clientMessage: cf.legacy.node.ClientActionMessage,
    instructionExecutor: InstructionExecutor
```

ideally it should just be 2, `instructions` and `clientMessage`